lib/iwdp.js

```diff
@@ -0,0 +1,78 @@
+import { SubProcess } from 'teen_process';
+import EventEmitter from 'events';
+import { execSync } from 'child_process';
+import Promise from 'bluebird';
+import logger from './logger';
+
+const request = Promise.promisify(require('request'));
+
+const IWDP_CMD = 'ios_webkit_debug_proxy';
+const MAX_RETRIES = 10;
+
+class IWDP extends EventEmitter {
+
+  constructor (udid = null) {
```

Comment:
We need to be able to support arbitrary ports for this, passed to the ios driver through `opts. webkitDebugProxyPort` (see https://github.com/appium/appium/blob/master/lib/parser.js#L345).